### PR TITLE
#2079 refactor FtpSinkTask to extend CsvSinkTask

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2079)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala [hunk 1]
```diff
@@ -99,6 +100,21 @@ class FtpStorage(ftpClient: FtpClient) extends Storage {
     */
   override def delete(path: String): Unit = if (exists(path)) ftpClient.delete(path)
 
+  @VisibleForTesting
+  def delete(path: String, recursive: Boolean): Unit = {
```
Comment (chia7712) [inline ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala]: please narrow down the scope of this method if it is exposed for testing.

File: ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala [hunk 2]
```diff
@@ -99,6 +100,21 @@ class FtpStorage(ftpClient: FtpClient) extends Storage {
...
+    if (exists(path)) {
+      if (recursive) {
+        ftpClient
+          .listFileNames(path)
```
Comment (chia7712) [inline ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala]: Does it cause exception if input path is file?

File: ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala (hunk 1 above)
Comment (eechih) [reply ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala]: ok

File: ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala (hunk 2 above)
Comment (eechih) [reply ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala]: It's OK to delete a file. I will write more test cases.

Comment (chia7712) [reply ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala]: I'm not sure FtpClient.listFileNames is able to accept a "file" path :(

Comment (eechih) [reply ohara-connector/src/main/scala/com/island/ohara/connector/ftp/FtpStorage.scala]: > I'm not sure FtpClient.listFileNames is able to accept a "file" path
Thanks. 
I have confirmed FtpClient.listFileNames doesn't accept a "file" path, so I will check the path is a folder before call it.

**Comment by eechih** [comment]
> Another Q: Is there a issue to introduce csv connector? It is worth having a document :)

Ok, I will fire a issue to write a document to introduce csv connector.

**Comment by eechih** [comment]
reopen to #2211

**Pull request closed without merging**